spotbugs-tests/src/test/java/edu/umd/cs/findbugs/nullness/TestCheckerFrameworkTypeAnnotations.java

```diff
@@ -16,11 +17,69 @@ public class TestCheckerFrameworkTypeAnnotations extends AbstractIntegrationTest
     @Test
     public void test() {
         performAnalysis("type/annotation/CheckerFrameworkTypeAnnotations.class");
-        final BugInstanceMatcher matcher = new BugInstanceMatcherBuilder()
+
+        // A matcher for all NP_NONNULL_RETURN_VIOLATION
+        final BugInstanceMatcher matcherMethodReturns = new BugInstanceMatcherBuilder()
                 .bugType("NP_NONNULL_RETURN_VIOLATION")
                 .build();
 
-        assertThat(getBugCollection(), containsExactly(1, matcher));
+        // Per-method NP_NONNULL_RETURN_VIOLATION matchers
+        final BugInstanceMatcher matcherReturningNullOnNonNullMethod = new BugInstanceMatcherBuilder()
+                .bugType("NP_NONNULL_RETURN_VIOLATION")
+                .inMethod("returningNullOnNonNullMethod")
+                .build();
+
+        final BugInstanceMatcher matcherReturningNullWithNonNullOnGenerics = new BugInstanceMatcherBuilder()
```

Comment:
IMO it's better if the positive tests are as specific as possible, and contain any additional info, which is added to the bug instance (e.g. source line location, variable).

Reply:
I've added the line number, let me know if that works for you